polkadot/runtime/parachains/src/assigner_on_demand/mod.rs

```diff
@@ -297,10 +434,83 @@ pub mod pallet {
 	}
 }
 
+// Internal functions and interface to scheduler/wrapping assignment provider.
 impl<T: Config> Pallet<T>
 where
 	BalanceOf<T>: FixedPointOperand,
 {
+	/// Take the next queued entry that is available for a given core index.
+	/// Invalidates and removes orders with a `para_id` that is not `ParaLifecycle::Parathread`
+	/// but only in [0..P] range slice of the order queue, where P is the element that is
+	/// removed from the order queue.
```

Comment:
the bit about `ParaLifecycle::Parathread` seems to be outdated in this PR - don't see any checks for paralifecycle here